### ISSUE TITLE
How to scan to customized types correctly.

### ISSUE THREAD

**Issue by nanoric** [eliaskosunen/scnlib#15]
Scanning one variable into customized types is described in test/usertype.cpp.
But what if I want to scan multiple values?
If there is one user_type inside arguments, the indexing will be wrong.
```cpp
# usertype.cpp #68

        std::string s1;
        std::string s2;
        auto ret = scn::scan("123 [4, 20] 1234", "{} {} {}", s1, ut, s2);
        CHECK(ret);
        CHECK(ut.val1 == 4);
        CHECK(ut.val2 == 20);
        CHECK(s1 == "123");
        CHECK(s2 == "1234");

```
excepted: test pass.
current:
```
[doctest] doctest version is "2.3.4"
[doctest] run with "--help" for options
===============================================================================
..\..\..\test\usertype.cpp(90):
TEST CASE:  user type<user_type>
  regular

..\..\..\test\usertype.cpp(75): ERROR: CHECK( s2 == "1234" ) is NOT correct!
  values: CHECK(  == 1234 )

===============================================================================
..\..\..\test\usertype.cpp(90):
TEST CASE:  user type<user_type2>
  regular

..\..\..\test\usertype.cpp(75): ERROR: CHECK( s2 == "1234" ) is NOT correct!
  values: CHECK(  == 1234 )

===============================================================================
[doctest] test cases:      3 |      1 passed |      2 failed |      0 skipped
[doctest] assertions:     21 |     19 passed |      2 failed |
[doctest] Status: FAILURE!
```

**Comment by eliaskosunen** [comment]
Somewhat fixed in 16ae919, although this should probably be fixed properly on the library side, rather than inserting a hack into the user customization code. For that reason, I'm keeping this open.

**Comment by eliaskosunen** [comment]
With the upcoming source range overhaul in 0.4 (also in current master), this is handled as follows:
https://github.com/eliaskosunen/scnlib/blob/df24d4089f63926e1a579a23b71703805222166c/test/usertype.cpp#L29-L38

Closing.
